### PR TITLE
revise discrete dp

### PR DESCRIPTION
**Pull request by arnavs**
Clean up per https://github.com/QuantEcon/lecture-source-jl/issues/439.

### CODE REVIEW

File: rst_files/discrete_dp.rst [hunk 1]
```diff
@@ -597,9 +642,10 @@ What happens if the agent is more patient?
 
 .. code-block:: julia
 
-    g_2 = SimpleOG(β=0.99)
+    g_2 = SimpleOG(β=0.99);
+    matrices_2 = QR_matrices(g_2);
```
Comment (jlperla) [inline rst_files/discrete_dp.rst]: I don't like this name.  Call it what it is, not the data structure.  `transition_matrices(...)` for example.
Also, I would prefer either splatting directly into the Q and R matrices, or forgetting the named tuple and just returning multiple arguments.

Notice
```
julia> using Parameters

julia> nt = (a = 2, b = 2)
(a = 2, b = 2)

julia> @unpack a, b = nt
(a = 2, b = 2)

julia> a
2

julia> b
2

julia> f() = (a = 2, b = 2)
f (generic function with 1 method)

julia> a, b = f()
(a = 2, b = 2)

julia> a
2

julia> b
2
```
i.e. you can return a named tuple into multiple arguments without an unpack.  But if you are only using a simple function once or twice, you can just return the tuple.

File: rst_files/discrete_dp.rst [hunk 2]
```diff
@@ -597,9 +642,10 @@ What happens if the agent is more patient?
...
 
-    ddp_2 = DiscreteDP(g_2.R, g_2.Q, g_2.β)
+    ddp_2 = DiscreteDP(matrices_2.R, matrices_2.Q, g_2.β)
```
Comment (jlperla) [inline rst_files/discrete_dp.rst]: after the above, just use R and Q directly.

**Comment by arnavs** [comment]
OK @jlperla see now.

**Comment by arnavs** [comment]
Merging as Jesse's feedback is now incorporated.

**Pull request merged**